### PR TITLE
User API (Add Logger + Fix spacings)

### PR DESCRIPTION
**Pull request by iamrakesh28**
Fixes convict-git#100

### CODE REVIEW

**Comment by iamrakesh28** [comment]
User API is completed and can be merged.

File: src/processing/UserUtil.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+     */
+    public static void stopBoardSession() {
+    	/**
+    	 * Informs the server about closing the session of the user
+    	 * 
+    	 * identifier is "StopConnection"
+    	 * message is userId
+    	 * destination is the server's address
+    	 */
+    	ClientBoardState.communicator.send(
```
Comment (himanshu2047) [inline src/processing/UserUtil.java]: need to call stop from IServerCommunication instead of you sending.

Comment (iamrakesh28) [reply src/processing/UserUtil.java]: Done

**Comment by iamrakesh28** [comment]
The PR has both the classes for User APIs with added logger. It can be reviewed and merged.

**Comment by iamrakesh28** [comment]
Fixes convict-git#100

**Review by himanshu2047 — APPROVED** [review]

File: src/processing/CommunicateChange.java [hunk 2]
```diff
@@ -59,8 +62,25 @@ public static void provideChanges(
...
+		if (handler == null) {
+			logger.log(
+					ModuleID.PROCESSING, 
+					LogLevel.ERROR, 
+					"[#" + Thread.currentThread().getId() + "] "
+					+ "Couldn't find the handler for the identifier"
+			);
+			return;
+		}
+
```
Comment (RathoreDevansh08) [inline src/processing/CommunicateChange.java:74]: you can throw exception?

File: src/processing/UserUtil.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    	);
+    	return maxOccUser;
+    }
+    
+    /**
+     * Stops the board session for the user
+     */
+    public static void stopBoardSession() {
+    	
+    	IServerCommunication serverComm = new ServerCommunication();
```
Comment (RathoreDevansh08) [inline src/processing/UserUtil.java]: single space indentation? tab indentation should be at least of 4 spaces. Please check and correct it.

File: src/processing/UserUtil.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+     * 
+     * @param positions list of positions
+     * @return username
+     * @return null if there is no object present at those positions
+    */
+    
+    public static String getUser(ArrayList <Position> positions) {
+    	
+    	// null object
+    	if (positions == null) {
```
Comment (RathoreDevansh08) [inline src/processing/UserUtil.java]: single space indentation? SHould be at least of 4 spaces tab.

File: src/processing/UserUtil.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	 * @param boardId Board ID	
+	 */
+    public static void initiateUserConnection(
+    		String username, 
+    		String ipAddress, 
+    		String boardId
+    ) {
+    	
+    	// stores the information
+    	ClientBoardState.username  = new Username(username);
```
Comment (RathoreDevansh08) [inline src/processing/UserUtil.java]: nested indentation varies by tab of width of minimum 4 spaces. Please check and correct it.

Comment (RathoreDevansh08) [reply src/processing/UserUtil.java]: ....everywhere

**Review by RathoreDevansh08 — APPROVED** [review]

**Pull request merged**